packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -229,14 +299,61 @@ class Context {
         nativeAssetsPath,
         xcodeFrameworksDir,
       );
+      if (platform == TargetPlatform.macos &&
+          expandedCodeSignIdentity != null &&
+          expandedCodeSignIdentity.isNotEmpty &&
+          environment['CODE_SIGNING_REQUIRED'] != 'NO') {
+        for (final FileSystemEntity entity in nativeAssetsDir.listSync()) {
+          if (entity is Directory) {
+            final String? frameworkName = parseFrameworkNameFromDirectory(entity);
+            if (frameworkName != null) {
+              _codesignFramework(
```

Comment:
>   /// On macOS, also codesigns the framework binaries. Codesigning occurs here rather
>   /// than during the Run Script `build` phase because the `EXPANDED_CODE_SIGN_IDENTITY`
>   /// is not passed in the build settings during the `build` phase for macOS.

Is this an Xcode difference between iOS and macOS?

Ideally it would be codesigned during `assemble` in a `Target`. The `Target` build system is very nice in that it can fingerprint the inputs and outputs and skip steps that aren't needed.

Reply:
This is due to the way macOS is setup. On iOS the `build` phase is part of the Runner target:

![Screenshot 2025-05-02 at 11 49 30 AM](https://github.com/user-attachments/assets/b761f723-24e9-4cfd-8419-be99693a883b)

However, on macOS, it's part of the `Flutter Assemble` target, which is an Aggregate target that does not have code signing settings:

![Screenshot 2025-05-02 at 11 50 22 AM](https://github.com/user-attachments/assets/bc13ef2e-60aa-411a-b789-1146bf182a41)